42free.sh

```diff
@@ -25,8 +25,11 @@ stderr=""
 current_dir=$(pwd)
 script_dir="$HOME/.scripts"
 script_path="$script_dir/42free.sh"
-sgoinfre_root="/sgoinfre/goinfre/Perso/$USER"
-sgoinfre_alt="/nfs/sgoinfre/goinfre/Perso/$USER"
+    # Find the first match of the corresponding user 
+sgoinfre_searched_path="$(find /sgoinfre/ -type d -name "$USER" -print -quit 2>/dev/null | grep "$USER")"
+    # Store only the path until the first match of $USER + add $USER
+sgoinfre_root=$(awk -v var="$USER" -F"${USER}" '{print $1 var}' <<< "$sgoinfre_searched_path")
```

Comment:
How about:
```bash
sgoinfre_root=$(find /sgoinfre/ -type d -name "$USER" -print -quit | grep -oE "^.*$USER" | head -n 1) 2>/dev/null
```

If that works, could you replace lines 28-31 with just that line?